### DISCUSSION TITLE
Pagination display error when call links() 2 times

### DISCUSSION THREAD

**Discussion by nguyentranchung** [livewire/livewire#3144] [7. Bugs (archived)] [answered]
![image](https://user-images.githubusercontent.com/9611224/93036402-ba48d600-f67a-11ea-9c37-59d2db363162.png)

Related #1536

**Comment by imClement** [comment]
I can confirm the same problem (Jetstream 1.0.2, Livewire 2.1.2).

For my install this problem only occurs when ```{{ $posts->links() }}``` is used **both** above and bellow the datatable
```
{{ $posts->links() }}
.....
@foreach
.......
@endforeach
.....
{{ $posts->links() }}
```

**Comment by imClement** [comment]
PS - Another problematic behavior occurs with the following steps:

1. Click on  the last page number in the list
2. Click on the first page number in the list

At that point, the total number of items in the list is not correct.

eg
```Showing 1 to 644 results```

instead of 

```Showing 1 to 6 of 44 results```

**Comment by lotfi-abdallah** [comment]
I'm having the same problem

- Laravel v8.0
- Livewire v2.0
- tailwindcss

**Comment by calebporzio** [comment]
Please verify this issue occurs on the latest version of Livewire : 2.2.3

**Comment by imClement** [comment]
@calebporzio 

The problem with multiplying and doubling the page numbers in Pagination **is still there** in 2.2.3 
Ref: https://github.com/livewire/livewire/issues/1582#issuecomment-692083275


The problem with the wrong display of current items interval and the total **is not there** anymore in  2.2.3
Ref: https://github.com/livewire/livewire/issues/1582#issuecomment-692141964

**Comment by mo3auya91** [comment]
- Laravel v8.0
- Livewire v2.0
- tailwindcss

same problem when add localization to routes,
all livewire ajax request are post request and they are not localized
maybe you'r useing the routes like this "/livewire/message/component-name" instead of using the roues names
like in jetstream views, I replaced many jetstream routes with routes names to get them work.

when I click page 2 this what happend

<img width="1440" alt="Screenshot 2020-09-15 at 23 16 49" src="https://user-images.githubusercontent.com/6853643/93260092-8fa96b00-f7a9-11ea-8bd0-e1d7c9c6e3d8.png">
I think its related to this issue #765

**Comment by breadthe** [comment]
I'm having a similar issue. After "live" filtering a datatable on one of the columns, the top set of links shows the number of pages (10) for the unfiltered data (which is wrong), while the bottom set displays the correct pages (5) for the filtered results.

If I click a page that's out of range, such as 10, it says "Showing to of 81 results", but does not show any results. However both top & bottom page sections now display the correct number of pages (5).

One other thing - I put the filtered value in the query string using `protected $queryString = ['filter'];`

Using Laravel 8.6 & Livewire 2.2.6.

![2020-09-23-livewire-pagination-bug](https://user-images.githubusercontent.com/17433578/94098145-978e8e00-fded-11ea-909c-d17916a95dcb.jpg)

**Comment by schwampf** [comment]
Having the same Problem. Did anybody find a workaround?

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by reliq** [comment]
REOPEN

**Comment by breadthe** [comment]
REOPEN

**Comment by tvarwig** [comment]
![Screen Shot 2020-11-11 at 10 31 05 AM](https://user-images.githubusercontent.com/8144988/98837905-29d40b00-2409-11eb-8efb-27f74a7df921.png)

I am having the same issue as seen above.

"livewire/livewire": "^2.3"
 "laravel/framework": "^8.12"

**Comment by Smef** [comment]
I solved this by using a custom pagination links view and adding a randomly generated ID for the keys of the wire:key. The 

At the top:

```
@php
    $paginatorId =  uniqid('paginator');
@endphp

```
and then replace
`<span wire:key="paginator-page{{ $page }}">`
with 
`<span wire:key="paginator-page{{ $page . '-' . $paginatorId }}">`

As an additional bonus, you can add to the top level div to give it an ID if it's passed in the links() function as a parameter (thanks to @likeadeckofcards for that)

`<div @if($id ?? false)id="{{$id}}"@endif>
`

**Comment by schwampf** [comment]
> 
> 
> I solved this by using a custom pagination links view and adding a randomly generated ID for the keys of the wire:key. The
> 
> At the top:
> 
> ```
> @php
>     $paginatorId =  uniqid('paginator');
> @endphp
> ```
> 
> and then replace
> `<span wire:key="paginator-page{{ $page }}">`
> with
> `<span wire:key="paginator-page{{ $page . '-' . $paginatorId }}">`
> 
> As an additional bonus, you can add to the top level div to give it an ID if it's passed in the links() function as a parameter (thanks to @likeadeckofcards for that)
> 
> `<div @if($id ?? false)id="{{$id}}"@endif> `

Great job, solved the problem for me.

Cheers

**Comment by austenc** [comment]
Hey folks, I'd love to solve this issue for you, but I'm not able to replicate it. Can someone please provide a public repo or some specific reproduction code? 

If you can do that, I'll figure out how to fix it. Thank you!

**Comment by squishythejellyfish** [comment]
👋 Oh Hi! I'm Squishy, the friendly jellyfish that manages Livewire issues.

I see this issue has been closed.

Here in the Livewire repo, we have an "issues can be closed guilt-free and without explanation" policy.

If for ANY reason you think this issue hasn't been resolved, PLEASE feel empowered to re-open it.

Re-opening actually helps us track which issues are a priority.

Reply "REOPEN" to this comment and we'll happily re-open it for you!

(More info on this philosophy here: https://twitter.com/calebporzio/status/1321864801295978497)

**Comment by jianjye** [comment]
This issue still happens in Livewire 2.4.4 unfortunately. Workaround I used is by setting a uniqid per the suggestion in this thread. Hope it gets fixed soon. 

```
<div wire:key="{{ uniqid('paginator') }}-page{{ $page }}">
    {{ $organizations->links() }}
</div>
```

**Comment by reliq** [comment]
REOPEN

**Comment by yakoffka** [comment]
reopen

**Comment by ChrisThompsonTLDR** [comment]
This seems to be an issue when multiple calls to the same `$models->links()` is called on the same page.

**Reply by joshhanley** [reply]
@ChrisThompsonTLDR this was fixed in v2.6. So make sure to update to the latest version. If you have published the pagination views though, you will need to republish them to get the latest ones for it to work correctly.

**Comment by lotfi-abdallah** [comment]
I wonder how big laravel dashboard frameworks like filament solve this problem

**Reply by reliq** [reply]
Would be funny to find out they don't

**Reply by joshhanley** [reply]
@Lootfi this was fixed in v2.6. So make sure to update to the latest version. If you have published the pagination views though, you will need to republish them to get the latest ones for it to work correctly.

**Reply by lotfi-abdallah** [reply]
That's true, this should be closed then. Thanks for the hard work.

**Reply by joshhanley** [reply]
No worries! 🙂

**Comment by joshhanley** [comment, marked as answer]
This was fixed in v2.6. So make sure to update to the latest version. If you have published the pagination views though, you will need to republish them to get the latest ones for it to work correctly.
